src/Psalm/Internal/Analyzer/ClassAnalyzer.php

```diff
@@ -2627,5 +2635,119 @@ private function checkEnum(PhpParser\Node\Stmt\Enum_ $enum): void
                 }
             }
         }
+
+        $case_values = [];
+        $case_names = [];
+
+        foreach ($enum->stmts as $stmt) {
+            if ($stmt instanceof PhpParser\Node\Stmt\EnumCase) {
+                if (isset($case_names[$stmt->name->name])) {
+                    if (IssueBuffer::accepts(
+                        new DuplicateEnumCase(
+                            'Enum case names should be unique',
+                            new CodeLocation($this, $stmt),
+                            $enum->name->name
+                        )
+                    )) {
+                        continue;
+                    }
+                }
+
+                $case_names[$stmt->name->name] = true;
+
+                if (null !== $stmt->expr && null === $enum->scalarType) {
+                    if (IssueBuffer::accepts(
+                        new InvalidEnumCaseValue(
+                            'Case of a non-backed enum should not have a value',
+                            new CodeLocation($this, $stmt),
+                            $enum->name->name
+                        )
+                    )) {
+                        continue;
+                    }
+                } elseif (null === $stmt->expr && null !== $enum->scalarType) {
+                    if (IssueBuffer::accepts(
+                        new InvalidEnumCaseValue(
+                            'Case of a backed enum should have a value',
+                            new CodeLocation($this, $stmt),
+                            $enum->name->name
+                        )
+                    )) {
+                        continue;
+                    }
+                } elseif (null !== $enum->scalarType && null !== $stmt->expr) {
+                    if (ExpressionAnalyzer::analyze(
+                        new StatementsAnalyzer(
+                            $this,
+                            $type_provider = new \Psalm\Internal\Provider\NodeDataProvider()
+                        ),
+                        $stmt->expr,
+                        $enum_context,
+                        false,
+                        $global_context,
+                        true
+                    ) === false) {
+                        // xxx: should we rather return here?
+                        continue;
+                    }
+
+                    $case_type = $type_provider->getType($stmt->expr);
+
+                    if (null === $case_type) {
+                        // xxx: can this even happen?
```

Comment:
I've seen it happen a lot when building plugins but I had no way to see what happened in the actual analysis so it may have been case of UndefinedClass like above...